src/Carousel.js

```diff
@@ -0,0 +1,149 @@
+import React, { useState, useEffect, useRef } from 'react';
+import './Carousel.css';
+
+function Carousel() {
+  let SliderArr = ['Первый', 'Второй', 'Последний'];
+
+  const [x, setX] = useState(0);
+  const [startX, setStartX] = useState(0);
+  const [offsetX, setOffsetX] = useState(0);
+
+  const [mouseDown, setMouseDown] = useState(false);
+
+  const widthOfTheSlide = 100;
+  const standardOfDifference = 50;
```

Comment:
If the constants do not change, then also put it out of the component and write with a large, like a define